pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -738,6 +765,258 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 			})
 		})
 	})
+
+	Context("validating admission policy tests", func() {
+
+		var (
+			policyBinding *admissionregistrationv1.ValidatingAdmissionPolicyBinding
+			policy        *admissionregistrationv1.ValidatingAdmissionPolicy
+		)
+
+		BeforeEach(func() {
+			By("Unmarshalling the VAP/VAPB yamls")
+
+			clusterAPIAdmissionPolicies := &corev1.ConfigMap{}
+			clusterAPICustomAdmissionPolicies := &corev1.ConfigMap{}
+
+			policyBinding = &admissionregistrationv1.ValidatingAdmissionPolicyBinding{}
+			policy = &admissionregistrationv1.ValidatingAdmissionPolicy{}
+
+			configMaps, err := os.Open("../../../manifests/0000_30_cluster-api_09_admission-policies.yaml")
+			Expect(err).ToNot(HaveOccurred())
+
+			// Order is important as the yaml decoder.Decode unmarshals
```

Comment:
Given we likely expect to ship a bunch of across the `cluster-api-custom-admission-policies` config map, I think relying on ordering will become increasingly brittle. 

Maybe we can decode into a client.Object, read stream, and look for the the `type` and `name` that we want?

cc @damdo

Reply:
Agree, good idea!

Reply:
Could also store things in a fake client / map to make getting easy